warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/partition/SortedTabletLocationPartitioner.java

```diff
@@ -0,0 +1,100 @@
+package datawave.ingest.mapreduce.partition;
+
+import com.google.common.collect.TreeMultimap;
+import datawave.ingest.mapreduce.job.SplitsFileType;
+import org.apache.hadoop.io.Text;
+import org.apache.log4j.Logger;
+
+import java.util.*;
+
+public class SortedTabletLocationPartitioner extends MultiTableRangePartitioner {
+
+    private static final Logger log = Logger.getLogger(SortedTabletLocationPartitioner.class);
+    private final Map<String,Map<Integer,Integer>> SPLIT_TO_REDUCER_MAP = new HashMap<>();
+
+    @Override
+    protected int calculateIndex(int index, int numPartitions, String tableName, int cutPointArrayLength) {
+        if(isAssignedPartition(tableName, index)){
+
+        }
+        assignPartitions(numPartitions, tableName, cutPointArrayLength);
```

Comment:
should this only be called if the partitions haven't yet been assigned?

Reply:
You're right. With the second if block replacing the first,  assignPartitions will only be called if isAssignedPartition is false